app/src/backend/HiapService.ts

```diff
@@ -0,0 +1,466 @@
+          jobStatus = HighImpactActionRankingStatus.SUCCESS;
+          break;
+        case "failed":
+          jobStatus = HighImpactActionRankingStatus.FAILURE;
+          await ranking.update({
+            status: HighImpactActionRankingStatus.FAILURE,
+          });
+          throw new Error("Prioritization job failed");
+        default:
+          jobStatus = HighImpactActionRankingStatus.PENDING;
+          break;
+      }
+      pollCount++;
+    }
+    // Fetch result
+    const actionRanking: PrioritizerResponse =
+      await getPrioritizationResult(jobId);
+
+      // Merge and save ranked actions with details for this language
+    const rankedActions = [
+      ...actionRanking.rankedActionsMitigation.map((a) => ({ ...a, type: ACTION_TYPES.Mitigation })),
+      ...actionRanking.rankedActionsAdaptation.map((a) => ({ ...a, type: ACTION_TYPES.Adaptation })),
+    ];
+    const mergedRanked = await fetchAndMergeRankedActions(lang, rankedActions);
+    await saveRankedActionsForLanguage(ranking, mergedRanked, lang);
+
+    await ranking.update({ status: HighImpactActionRankingStatus.SUCCESS });
+    return ranking;
+  } catch (err) {
+    logger.error({ err }, "Error in runActionRankingJob");
+  }
+};
+
+async function getCityContextAndEmissionsData(
+  inventoryId: string,
+): Promise<PrioritizerCityData> {
+  // Get inventory to access city information
+  const inventory = await db.models.Inventory.findByPk(inventoryId, {
+    include: [{ model: db.models.City, as: "city" }],
+  });
+  if (!inventory || !inventory.city) {
+    throw new Error("Inventory or city not found");
+  }
+
+  const locode = inventory.city.locode;
+  if (!locode) {
+    throw new Error("City locode not found");
+  }
+
+  const population = await PopulationService.getPopulationDataForCityYear(
+    inventory.city.cityId,
+    inventory.year!,
+  );
+  const emissions: EmissionsBySector[] = await getTotalEmissionsBySector([
+    inventory.inventoryId,
+  ]);
+  
+  const cityData = {
+    cityContextData: {
+      locode: locode,
+      populationSize: population.population || null,
+    },
+    cityEmissionsData: {
+      stationaryEnergyEmissions:
+        Number(
+          emissions.find((e) => e.sector_name === "Stationary Energy")?.co2eq,
+        ) ?? null,
+      transportationEmissions:
+        Number(
+          emissions.find((e) => e.sector_name === "Transportation")?.co2eq,
+        ) ?? null,
+      wasteEmissions:
+        Number(emissions.find((e) => e.sector_name === "Waste")?.co2eq) ?? null,
+      ippuEmissions:
+        Number(
+          emissions.find(
+            (e) =>
+              e.sector_name === "Industrial Processes and Product Uses (IPPU)",
+          )?.co2eq,
+        ) ?? null,
+      afoluEmissions:
+        Number(
+          emissions.find(
+            (e) =>
+              e.sector_name ===
+              "Agriculture, Forestry, and Other Land Use (AFOLU)",
+          )?.co2eq,
+        ) ?? null,
+    },
```

Comment:
Can we extract the logic to find the emissions for a given sector name and convert to a number etc. here?